### PR TITLE
Additionalargs

### PR DESCRIPTION
**Pull request by smithed**
I figured I'd offer these changes up. Short version is that if you look at 688ef8d on Jan 16, I added a few new commands:
First is "VIA" which is the same as "VI" except is passes any extra arguments to an array-of-strings control called "args" on the front panel of the called vi.
Second is "VIPB" which is a vastly more complicated version of "VIP" and includes installing dependencies, making sure the code to be built is uninstalled, and then installing and uploading the resultant package to a repository.

if none of these appeal feel free to ignore, but its freshly pulled from your code so you shouldn't have any issues merging.

### CODE REVIEW

**Comment by smithed** [comment]
oh, except you literally just made changes :(

**Comment by BarrettStephen** [comment]
ah man.... how unfortunate timing. would have been nice to get these easily

btw you should squash your commits before pushing them upstream

**Comment by smithed** [comment]
how does one do that, and why would you want to? I mean in this case sure since the commits are nonsensical but in some situations I actually try and don't want to lose that history.

**Comment by smithed** [comment]
Also, what changes had you been making? When I opened the code I didn't see much different even on the things marked as changed. The only thing I noticed was an additional VI input which would be easy to pull into mine. On the other hand most of mine is self-contained as well so if there were major changes I could try integrating them from scratch. I'd rather not have two copies of the same code :)

**Comment by BarrettStephen** [comment]
You want to squash commits before you push because it avoids cluttering up history with changes that didn't matter, it aids in bisecting, and it aids in reverting. Each commit should be a feature, so you can roll back a feature by rolling back a single commit. With a ton of commits for a single feature there isn't a clear line of when something started and finished. This is very easy to do with TortoiseGit, simply go to the working directory... right click -> git -> show log. Then highlight the commits you want to combine, right click -> combine into one commit. 

When you do this, it will combine your commit messages as well, you should clean this up into a single easy to read feature commit message. You can even retroactively do this... just force the push (when you're finished combining) so it overwrites the history on the server. do this with the check box 'force may discard known changes'. I actually just did that to combine two commits on my repo. 
For more how-to-use-git stuff here is an NI internal presentation
http://ni-tech.natinst.com/details.php?idPres=2809

Here is the diff of my change:
https://github.com/NIVeriStandAdd-Ons/Jenkins-Autobuild/commit/20d51a919880add6b3ee871ab91c9285e1110f20

Basically my change just adds in a new parameter. To implement this I:
1. Updated build.vi to read the new parameter into the cluster and to use a new typedef for command line args named "Command Line Args.ctl". I also moved the parameter 'Source Ver' into the command line args ctl instead of the top level control like it was before
2. Updated Build Action.vi to take Command Line Args.ctl
3. Updated Run VI to pass max build ver to the VI
4. Updated the batch files with significantly more comments about the parameters and to add in the new maxThisBuildVer parameter

If you still want to merge this pull request in, then typical git practices would be that you rebase your code and resubmit the pull request after doing so. Which would be 10000x easier if this was all text based...

**Comment by smithed** [comment]
If this were text based I also probably wouldn't have any commits that say "I dunno, labview changed something". anyway, I'll take a look.
I'll resubmit when ive made the changes
